velox/exec/tests/HashJoinTest.cpp

```diff
@@ -206,6 +246,28 @@ TEST_F(HashJoinTest, filter) {
       "((t_k0 % 100) + (u_k0 % 100)) % 40 < 20");
 }
 
+TEST_F(HashJoinTest, filterOrder) {
+  // In this join, the tables have different schema. The filter predicate uses
+  // a column from the right table before the left and the corresponding
+  // columns at the same channel number(1) have different types. This has been
+  // a source of errors in the join logic.
+  testJoinTablesDifferentSchemas(
```

Comment:
The name of this function doesn't match the test name. It might be simpler to not use a helper function. The test name reads as adaptive filter re-ordering, which is not what it does. Is there a better name?

The filter used in the query is pretty complex. Does it need to be such?

Reply:
Changed the name of the test.

I was trying to keep the original TPC-H 19 filter but you're right in that a simpler one could do for the purposes of this test. Have changed the filter.